### PR TITLE
EQL: Improve query folder test suite

### PR DESCRIPTION
**Pull request by aleksmaus**
Related to https://github.com/elastic/elasticsearch/issues/52775

@costin could you check and let me know if that's what you were expecting to be done for this issue.
I didn't add tests for the queries that are not supported yet, cause the errors are already tested in the verifier tests.

### CODE REVIEW

**Review by imotov — COMMENTED** [review]
As we expect these tests to grow, maybe it's a good time to invest into a better test infrastructure. Pasting 100 lines long json blobs into strings for comparison might not scale that well.

**Comment by aleksmaus** [comment]
That the next thing I'm working on: helper wrappers with test specs that can be shared across eql and HL client tests, as a part of https://github.com/elastic/elasticsearch/issues/52846.
This PR is primarily to gather the feedback on what was expected for this specific github issue.
The expectations were not quite clear to me.

**Comment by aleksmaus** [comment]
Talked to Costin about this PR, will be changing it back to substrings testing, pretty much the same how SQL query folder tests were done.

**Review by costin — APPROVED** [review]
LGTM

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+                () -> plan("process where (serial_event_id<9 and serial_event_id >= 7) or (opcode == pid)"));
+        String msg = e.getMessage();
+        assertEquals("Line 1:74: Comparisons against variables are not (currently) supported; offender [pid] in [==]", msg);
+    }
+
+    public void testPropertyEquationInClauseFilterUnsupported() {
+        QlIllegalArgumentException e = expectThrows(QlIllegalArgumentException.class,
+                () -> plan("process where opcode in (1,3) and process_name in (parent_process_name, \"SYSTEM\")"));
+        String msg = e.getMessage();
+        assertEquals("Line 1:52: Comparisons against variables are not (currently) supported; offender [parent_process_name] in [==]", msg);
```
Comment (astefan) [inline x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java]: This error message is confusing to the end user: there is no equality comparison in the query... I'm not saying the test is incorrect, but the way we report the error in the context of EQL is slightly incorrect.

**Review by astefan — COMMENTED** [review]
There is one test that has an error message check that seems a bit off and can be confusing to the users.

File: x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java (hunk 1 above)
Comment (aleksmaus) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java]: Agreed. @costin thoughts?

Comment (costin) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java]: I agree - to improve the message one could either look and check whether it's the first comparison (which might not work if `any` is added) or check whether the comparison is added against the event_type field and if so, mention that in the error message `(e.g. offender `eventy.category`).

Comment (aleksmaus) [reply x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/planner/QueryFolderFailTests.java]: Will open a separate issue for the error message per conversation with Costin.

**Pull request merged**